src/ray/raylet/node_manager.cc

```diff
@@ -519,6 +519,22 @@ ray::Status NodeManager::RegisterGcs() {
         [this] { local_object_manager_.FlushFreeObjects(); },
         RayConfig::instance().free_objects_period_milliseconds(),
         "NodeManager.deadline_timer.flush_free_objects");
+    periodical_runner_.RunFnPeriodically(
+        [this] {
+          // Trigger object spilling if current usage is above the specified threshold.
+          const float allocated_percentage =
+              static_cast<float>(local_object_manager_.GetPrimaryBytes()) /
+              object_manager_.GetMemoryCapacity();
+          if (allocated_percentage >= RayConfig::instance().object_spilling_threshold()) {
+            RAY_LOG(INFO) << "Triggering object spilling because current primary objects usage "
+                          << allocated_percentage * 100 << "% is above threshold "
+                          << RayConfig::instance().object_spilling_threshold() * 100
+                          << "%.";
+            local_object_manager_.SpillObjectUptoMaxThroughput();
+          }
+        },
+        RayConfig::instance().free_objects_period_milliseconds(),
```

Comment:
Do you know how frequently we triggered threashold-based spilling before this? Now it seems like we are doing once a second. Was it the same, or more frequent? Is it possible to preserve the same interval & frequency to minimze the impact of changes>?

Reply:
Before it was once per object. Good idea; I'll add the same check to the object added handler.